jax/_src/lax/lax.py

```diff
@@ -1109,12 +1122,27 @@ def _reduction_jaxpr(computation, aval):
   jaxpr, _, consts = pe.trace_to_jaxpr(comp, (pval, pval), instantiate=False)
   return jaxpr, consts
 
-def _get_monoid_reducer(monoid_op: Callable, x: Array) -> Optional[Callable]:
+@cache()
+def _variadic_reduction_jaxpr(computation, flat_avals, aval_tree):
+  avals = tree_util.tree_unflatten(aval_tree, flat_avals)
+  flat_in_avals, in_tree = tree_util.tree_flatten((avals, avals))
+  pvals = safe_map(pe.PartialVal.unknown, flat_in_avals)
+  comp = lu.wrap_init(computation)
+  flat_comp, out_tree = api_util.flatten_fun_nokwargs(comp, in_tree)
+  jaxpr, _, consts = pe.trace_to_jaxpr(flat_comp, tuple(pvals),
+                                       instantiate=False)
+  return jaxpr, consts, out_tree()
+
+def _get_monoid_reducer(monoid_op: Callable, xs: Array, **reduce_kwargs) -> Optional[Callable]:
+  if len(xs) > 1:
```

Comment:
Presumably you should test for exact equality. What happens if I attempt a nullary reduction?

(That may be illegal at the XLA level (we should check), but there's clearly nothing stopping me from getting this far with one.)

Reply:
Done.